internal/kafka/internal/services/kafka.go

```diff
@@ -309,8 +309,10 @@ func (k *kafkaService) reserveQuota(kafkaRequest *dbapi.KafkaRequest) (subscript
 			return "", errors.NewWithCause(errors.ErrorGeneral, err, "failed to count kafka %s instances", instType.DisplayName)
 		}
 
-		if count > 0 {
-			return "", errors.TooManyKafkaInstancesReached("only one %s instance is allowed", instType.DisplayName)
+		maxAllowedTrialInstances := k.kafkaConfig.Quota.MaxAllowedTrialInstance
+
+		if count > int64(maxAllowedTrialInstances) {
```

Comment:
I believe this should be `>=` to avoid allowing `maxAllowedTrialInstances + 1` instances.

Reply:
Is that correct?

The count attribute is the the number of already existing kafka developer instances.
For example, if there is 1 already existing kafka instance the count would be 1.
If you set maxAllowedTrialInstances to 1 then the condition would be
1 >= 1 which would be true, which means that creation would be allowed, which is not what we want.

Reply:
if count is 1, 1>=1 would be true, so it would enter the `if` and return an error

Reply:
+1. It'll be good to have/add a unit test for this as well.

Reply:
Yes, you are right. It should be `>=`. I'll make this change.

Reply:
> +1. It'll be good to have/add a unit test for this as well.

We have unit test for it. But it's failing. Can you suggest me what is wrong in test ? This is my first test in kfm and golang :(

https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/ce15e67a61a882a968291405074eb736053c6178/internal/kafka/internal/services/kafka_test.go#L1180

Reply:
You'll need to mock the `count` SQL query in the test.
Something like
```go
mocket.Catcher.NewMock().WithQuery(`SELECT count(1) FROM "kafka_requests" WHERE instance_type = $1 AND owner = $2 AND (organisation_id = $3) AND "kafka_requests"."deleted_at" IS NULL`).
					WithArgs(types.DEVELOPER.String(), testUser, "org-id").
					WithReply([]map[string]interface{}{{"count": 1}})
```
added before https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/ce15e67a61a882a968291405074eb736053c6178/internal/kafka/internal/services/kafka_test.go#L1229 will do.

Reply:
I'll also suggest to:
change the value of https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/ce15e67a61a882a968291405074eb736053c6178/internal/kafka/internal/services/kafka_test.go#L1190 to 2 and the returned mock count (see above comment) db query response to 2. 

Also, I'll suggest to add another test case that pass the max developer instance check.
i.e `MaxAllowedTrialInstance` is say maybe 1 and the count in the database is `0`. With this setup, the test should pass and able to create the developer kafka.